provisioning_templates/snippet/redhat_register.erb

```diff
@@ -100,6 +109,31 @@ snippet: true
     fi
   <% end %>
 
+  <%- if (host_param('syspurpose_role') || host_param('syspurpose_usage') || host_param('syspurpose_sla') || host_param('syspurpose_addons')) %>
+    <%- if !atomic %>
+      dnf -y install subscription-manager-syspurpose
+    <%- end %>
+
+    if [ -f /usr/sbin/syspurpose ]; then
```

Comment:
Should this be `-x`?

Reply:
I don't think so.  If something in bin or sbin isn't executable, but it should be, I think it's better if we let the shell print out the error that it can't execute it, rather than not trying at all.  It's also `-f` everywhere else